chain/chain/src/chain_update.rs

```diff
@@ -370,6 +370,12 @@ impl<'a> ChainUpdate<'a> {
                     apply_result.outcomes,
                     outcome_paths,
                 );
+                self.chain_store_update.save_state_transition_data(
```

Comment:
`apply_chunk_postprocessing` differentiates between `ShardUpdateResult::Stateful` and `ShardUpdateResult::Stateless`, where `process_apply_chunk_result` is only called for `ShardUpdateResult::Stateful`. Just wanted to confirm that we intentionally `save_state_transition_data` as a part of `process_apply_chunk_result`.

Reply:
Good question! That sounds correct because only on the stateful path would we want to persist anything. The stateless path is only for validation, the result of which is just to send a chunk endorsement and then the whole thing can be discarded.

That being said though, @Longarithm do we still need this code path of calling apply_chunk_postprocessing for statelessly validated chunks?, if we're going to do chunk validation in the chunk_validation.rs on a separate thread (as opposed to the chunk application thread that we have in the chain)? Seems better to not multiplex this code path as it's a lot more complicated and more things can go wrong.

Reply:
Great, more code to remove!